### PR TITLE
Add support for field selectors

### PR DESCRIPTION
**Pull request by palourde**
## What is this change?

This PR adds functions to retrieve fields for resources that can be listed via the REST API.

It also adds an argument to the lister HTTP handler that correspond to these functions signature.

## Why is this change necessary?

Required by https://github.com/sensu/sensu-enterprise-go/issues/283.

## Does your change need a Changelog entry?

Added!

## Do you need clarification on anything?

Nope!

## Were there any complications while making this change?

Nope! I hesitated to add a method to the `Resource` interface but I decided against it because it could be considered as a breaking change; third-party code relying on it would break with this change.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Not yet.

## How did you verify this change?

Manual testing.

### CODE REVIEW

**Comment by palourde** [comment]
CircleCI builds are just getting OOM...

File: api/core/v2/namespace.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func FixtureNamespace(name string) *Namespace {
 	}
 }
 
-// URIPath returns the path component of a Namespace URI.
+// URIPath returns the path component of a Namespace 	URI.
```
Comment (ccressent) [inline api/core/v2/namespace.go]: Minor nitpick: that whitespace diff doesn't look right.

File: backend/apid/routers/lister_test.go [hunk 2]
```diff
@@ -92,8 +92,8 @@ func TestList(t *testing.T) {
 			w := httptest.NewRecorder()
 
 			router := mux.NewRouter()
-			router.PathPrefix("/foo/{subcollection}").HandlerFunc(List(controller.List))
-			router.PathPrefix("/foo").HandlerFunc(List(controller.List))
+			router.PathPrefix("/foo/{subcollection}").HandlerFunc(List(controller.List, corev2.CheckConfigFields))
+			router.PathPrefix("/foo").HandlerFunc(List(controller.List, corev2.CheckConfigFields))
```
Comment (ccressent) [inline backend/apid/routers/lister_test.go]: I would suggest passing in `func(r corev2.Resource) map[string]string { return map[string]string{} }`, a "no-op" `FieldsFunc`, if you just need one for the sake of passing one in.

File: api/core/v2/asset.go [hunk 3]
```diff
@@ -101,3 +102,13 @@ func (a *Asset) URIPath() string {
 func NewAsset(meta ObjectMeta) *Asset {
 	return &Asset{ObjectMeta: meta}
 }
+
+// AssetFields returns a set of fields that represent that resource
+func AssetFields(r Resource) map[string]string {
+	resource := r.(*Asset)
+	fields := make(map[string]string, 3)
```
[the remark this answers is not in the document]
Comment (palourde) [reply api/core/v2/asset.go]: Even better, we will just use composite literals

**Pull request merged**